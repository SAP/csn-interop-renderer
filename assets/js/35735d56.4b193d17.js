"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[775],{2082:e=>{e.exports=JSON.parse('{"version":{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"noIndex":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"defaultSidebar":[{"type":"link","label":"Overview","href":"/csn-interop-renderer/","docId":"index","unlisted":false},{"type":"category","label":"Integrations","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"NPM package","href":"/csn-interop-renderer/Integrations/npm","docId":"Integrations/npm","unlisted":false},{"type":"link","label":"Web Component","href":"/csn-interop-renderer/Integrations/web-component","docId":"Integrations/web-component","unlisted":false},{"type":"link","label":"Using CLI","href":"/csn-interop-renderer/Integrations/cli","docId":"Integrations/cli","unlisted":false}]},{"type":"category","label":"examples","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Airline (Example)","href":"/csn-interop-renderer/examples/Airline","docId":"examples/Airline","unlisted":false},{"type":"link","label":"BillingDocument","href":"/csn-interop-renderer/examples/BillingDocument","docId":"examples/BillingDocument","unlisted":false},{"type":"link","label":"CostCenter","href":"/csn-interop-renderer/examples/CostCenter","docId":"examples/CostCenter","unlisted":false},{"type":"link","label":"CostCenterCombined","href":"/csn-interop-renderer/examples/CostCenterCombined","docId":"examples/CostCenterCombined","unlisted":false},{"type":"link","label":"GeneralLedgerAccountLineItem","href":"/csn-interop-renderer/examples/GeneralLedgerAccountLineItem","docId":"examples/GeneralLedgerAccountLineItem","unlisted":false}]}]},"docs":{"examples/Airline":{"id":"examples/Airline","title":"Airline (Example)","description":"This CSN example document shows how the airline example is expressed with a CDS Service exposing the entities through an API.","sidebar":"defaultSidebar"},"examples/BillingDocument":{"id":"examples/BillingDocument","title":"BillingDocument","description":"Entity Definitions","sidebar":"defaultSidebar"},"examples/CostCenter":{"id":"examples/CostCenter","title":"CostCenter","description":"Entity Definitions","sidebar":"defaultSidebar"},"examples/CostCenterCombined":{"id":"examples/CostCenterCombined","title":"CostCenterCombined","description":"Entity Definitions","sidebar":"defaultSidebar"},"examples/GeneralLedgerAccountLineItem":{"id":"examples/GeneralLedgerAccountLineItem","title":"GeneralLedgerAccountLineItem","description":"Entity Definitions","sidebar":"defaultSidebar"},"index":{"id":"index","title":"Overview","description":"Motivation","sidebar":"defaultSidebar"},"Integrations/cli":{"id":"Integrations/cli","title":"Using CLI","description":"Suitable for local testing on your machine.","sidebar":"defaultSidebar"},"Integrations/npm":{"id":"Integrations/npm","title":"NPM package","description":"Suitable for JavaScript/TypeScript based projects.","sidebar":"defaultSidebar"},"Integrations/web-component":{"id":"Integrations/web-component","title":"Web Component","description":"Suitable for Frontend e.g. React/Angular based projects.","sidebar":"defaultSidebar"}}}}')}}]);