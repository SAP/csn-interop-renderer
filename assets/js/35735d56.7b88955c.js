"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[775],{2082:e=>{e.exports=JSON.parse('{"version":{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"noIndex":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"defaultSidebar":[{"type":"link","label":"Overview","href":"/csn-interop-renderer/","docId":"index","unlisted":false},{"type":"category","label":"Integrations","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"NPM package","href":"/csn-interop-renderer/Integrations/npm","docId":"Integrations/npm","unlisted":false},{"type":"link","label":"Web Component","href":"/csn-interop-renderer/Integrations/web-component","docId":"Integrations/web-component","unlisted":false},{"type":"link","label":"Using CLI","href":"/csn-interop-renderer/Integrations/cli","docId":"Integrations/cli","unlisted":false}]},{"type":"category","label":"Renderer examples","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Airline","href":"/csn-interop-renderer/Renderer examples/Airline","docId":"Renderer examples/Airline","unlisted":false},{"type":"link","label":"BillingDocument","href":"/csn-interop-renderer/Renderer examples/BillingDocument","docId":"Renderer examples/BillingDocument","unlisted":false},{"type":"link","label":"CostCenter","href":"/csn-interop-renderer/Renderer examples/CostCenter","docId":"Renderer examples/CostCenter","unlisted":false},{"type":"link","label":"CostCenterCombined","href":"/csn-interop-renderer/Renderer examples/CostCenterCombined","docId":"Renderer examples/CostCenterCombined","unlisted":false},{"type":"link","label":"GeneralLedgerAccountLineItem","href":"/csn-interop-renderer/Renderer examples/GeneralLedgerAccountLineItem","docId":"Renderer examples/GeneralLedgerAccountLineItem","unlisted":false}]},{"type":"link","label":"Support","href":"/csn-interop-renderer/support","docId":"support","unlisted":false}]},"docs":{"index":{"id":"index","title":"Overview","description":"Motivation","sidebar":"defaultSidebar"},"Integrations/cli":{"id":"Integrations/cli","title":"Using CLI","description":"Suitable for local testing on your machine.","sidebar":"defaultSidebar"},"Integrations/npm":{"id":"Integrations/npm","title":"NPM package","description":"Suitable for JavaScript/TypeScript based projects.","sidebar":"defaultSidebar"},"Integrations/web-component":{"id":"Integrations/web-component","title":"Web Component","description":"Suitable for Frontend e.g. React/Angular based projects.","sidebar":"defaultSidebar"},"Renderer examples/Airline":{"id":"Renderer examples/Airline","title":"Airline","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/BillingDocument":{"id":"Renderer examples/BillingDocument","title":"BillingDocument","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/CostCenter":{"id":"Renderer examples/CostCenter","title":"CostCenter","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/CostCenterCombined":{"id":"Renderer examples/CostCenterCombined","title":"CostCenterCombined","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"Renderer examples/GeneralLedgerAccountLineItem":{"id":"Renderer examples/GeneralLedgerAccountLineItem","title":"GeneralLedgerAccountLineItem","description":"This Data Product definition description was done using the Core Schema Notation Interoperability (short: CSN Interop) format.","sidebar":"defaultSidebar"},"support":{"id":"support","title":"Support","description":"Reporting Bugs","sidebar":"defaultSidebar"}}}}')}}]);