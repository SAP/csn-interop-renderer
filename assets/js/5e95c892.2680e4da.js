"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[647],{8122:(__unused_webpack_module,__webpack_exports__,__webpack_require__)=>{eval('__webpack_require__.r(__webpack_exports__);\n/* harmony export */ __webpack_require__.d(__webpack_exports__, {\n/* harmony export */   "default": () => (/* binding */ DocsRoot)\n/* harmony export */ });\n/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(6540);\n/* harmony import */ var clsx__WEBPACK_IMPORTED_MODULE_4__ = __webpack_require__(4164);\n/* harmony import */ var _docusaurus_theme_common__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(1769);\n/* harmony import */ var _docusaurus_theme_common__WEBPACK_IMPORTED_MODULE_5__ = __webpack_require__(204);\n/* harmony import */ var _docusaurus_renderRoutes__WEBPACK_IMPORTED_MODULE_6__ = __webpack_require__(2831);\n/* harmony import */ var _theme_Layout__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(1410);\n/* harmony import */ var react_jsx_runtime__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(4848);\n/**\n * Copyright (c) Facebook, Inc. and its affiliates.\n *\n * This source code is licensed under the MIT license found in the\n * LICENSE file in the root directory of this source tree.\n */function DocsRoot(props){return/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_2__.jsx)(_docusaurus_theme_common__WEBPACK_IMPORTED_MODULE_3__/* .HtmlClassNameProvider */ .e3,{className:(0,clsx__WEBPACK_IMPORTED_MODULE_4__/* ["default"] */ .A)(_docusaurus_theme_common__WEBPACK_IMPORTED_MODULE_5__/* .ThemeClassNames */ .G.wrapper.docsPages),children:/*#__PURE__*/(0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_2__.jsx)(_theme_Layout__WEBPACK_IMPORTED_MODULE_1__/* ["default"] */ .A,{children:(0,_docusaurus_renderRoutes__WEBPACK_IMPORTED_MODULE_6__/* .renderRoutes */ .v)(props.route.routes)})});}//# sourceURL=[module]\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiODEyMi5qcyIsIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7OztBQUFBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQSxHQU1lLFFBQVMsQ0FBQVEsUUFBUUEsQ0FBQ0MsS0FBSyxDQUFFLENBQ3RDLG1CQUNFRixzREFBQSxDQUFDSixxRkFBcUIsRUFBQ08sU0FBUyxDQUFFVCx5REFBSSxDQUFDQyw4RUFBZSxDQUFDUyxPQUFPLENBQUNDLFNBQVMsQ0FBRSxDQUFBQyxRQUFBLGNBQ3hFTixzREFBQSxDQUFDRiw4REFBTSxFQUFBUSxRQUFBLENBQUVULCtFQUFZLENBQUNLLEtBQUssQ0FBQ0ssS0FBSyxDQUFDQyxNQUFNLENBQUMsQ0FBUyxDQUFDLENBQzlCLENBQUMsQ0FFNUIiLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly9Ac2FwL2Nzbi1pbnRlcm9wLXJlbmRlcmVyLXdlYnNpdGUvLi9ub2RlX21vZHVsZXMvQGRvY3VzYXVydXMvdGhlbWUtY2xhc3NpYy9saWIvdGhlbWUvRG9jc1Jvb3QvaW5kZXguanM/MjBlOCJdLCJzb3VyY2VzQ29udGVudCI6WyIvKipcbiAqIENvcHlyaWdodCAoYykgRmFjZWJvb2ssIEluYy4gYW5kIGl0cyBhZmZpbGlhdGVzLlxuICpcbiAqIFRoaXMgc291cmNlIGNvZGUgaXMgbGljZW5zZWQgdW5kZXIgdGhlIE1JVCBsaWNlbnNlIGZvdW5kIGluIHRoZVxuICogTElDRU5TRSBmaWxlIGluIHRoZSByb290IGRpcmVjdG9yeSBvZiB0aGlzIHNvdXJjZSB0cmVlLlxuICovXG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IGNsc3ggZnJvbSAnY2xzeCc7XG5pbXBvcnQge1RoZW1lQ2xhc3NOYW1lcywgSHRtbENsYXNzTmFtZVByb3ZpZGVyfSBmcm9tICdAZG9jdXNhdXJ1cy90aGVtZS1jb21tb24nO1xuaW1wb3J0IHJlbmRlclJvdXRlcyBmcm9tICdAZG9jdXNhdXJ1cy9yZW5kZXJSb3V0ZXMnO1xuaW1wb3J0IExheW91dCBmcm9tICdAdGhlbWUvTGF5b3V0JztcbmV4cG9ydCBkZWZhdWx0IGZ1bmN0aW9uIERvY3NSb290KHByb3BzKSB7XG4gIHJldHVybiAoXG4gICAgPEh0bWxDbGFzc05hbWVQcm92aWRlciBjbGFzc05hbWU9e2Nsc3goVGhlbWVDbGFzc05hbWVzLndyYXBwZXIuZG9jc1BhZ2VzKX0+XG4gICAgICA8TGF5b3V0PntyZW5kZXJSb3V0ZXMocHJvcHMucm91dGUucm91dGVzKX08L0xheW91dD5cbiAgICA8L0h0bWxDbGFzc05hbWVQcm92aWRlcj5cbiAgKTtcbn1cbiJdLCJuYW1lcyI6WyJSZWFjdCIsImNsc3giLCJUaGVtZUNsYXNzTmFtZXMiLCJIdG1sQ2xhc3NOYW1lUHJvdmlkZXIiLCJyZW5kZXJSb3V0ZXMiLCJMYXlvdXQiLCJqc3giLCJfanN4IiwiRG9jc1Jvb3QiLCJwcm9wcyIsImNsYXNzTmFtZSIsIndyYXBwZXIiLCJkb2NzUGFnZXMiLCJjaGlsZHJlbiIsInJvdXRlIiwicm91dGVzIl0sInNvdXJjZVJvb3QiOiIifQ==\n//# sourceURL=webpack-internal:///8122\n')}}]);