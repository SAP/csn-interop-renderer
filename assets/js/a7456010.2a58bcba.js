"use strict";(self.webpackChunk_sap_csn_interop_renderer_website=self.webpackChunk_sap_csn_interop_renderer_website||[]).push([[235],{8552:e=>{e.exports=JSON.parse('{"name":"docusaurus-plugin-content-pages","id":"default"}')}}]);