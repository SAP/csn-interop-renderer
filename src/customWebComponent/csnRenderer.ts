import { generateHtml } from "../index.js";

export interface CsnRendererProps {
  /** @param source A valid text (containing JSON CSNInteropRoot object).*/
  source: string;
}

type CsnRendererPropName = keyof CsnRendererProps;

export class CsnRenderer extends HTMLElement {
  private _htmlContent: string = "";
  private _source: string = "";

  public constructor() {
    super();
  }

  public static observedAttributes: CsnRendererPropName[] = ["source"];

  private async _renderHtml(value: string | null | undefined): Promise<void> {
    if (value) {
      this._htmlContent = await generateHtml(JSON.parse(value));
      this.innerHTML = this._htmlContent;
    }
  }

  public connectedCallback(): void {}

  public disconnectedCallback(): void {}

  public adoptedCallback(): void {}

  public attributeChangedCallback(name: CsnRendererPropName, _oldValue: unknown, newValue: unknown): void {
    if (name === "source" && !!newValue && typeof newValue === "string") {
      this._source = newValue;
      void this._renderHtml(this._source);
    }
  }
}

declare global {
  // registering the Angular typings of the custom element
  // eslint-disable-next-line @typescript-eslint/naming-convention
  interface HTMLElementTagNameMap {
    "csn-renderer": CsnRenderer;
  }

  // registering the React typings of the custom element
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace React.JSX {
    interface IntrinsicElements {
      ["csn-renderer"]: { source: string; config?: string };
    }
  }
}

customElements.define("csn-renderer", CsnRenderer);
