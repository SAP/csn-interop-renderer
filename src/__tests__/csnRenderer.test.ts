/**
 * @jest-environment jsdom
 */

import example from "../../examples/CSNInterop.js";
import { CsnRenderer } from "../customWebComponent/csnRenderer.js";

describe("Custom web-component Tests", () => {
  test("should properly render the provided CSN source document", async () => {
    const component = new CsnRenderer();
    component.setAttribute("source", JSON.stringify(example));
    // let some time pass until the async generateHtml function has finished
    await new Promise((result) => setTimeout(result, 10));
    expect(component).toBeDefined();
    expect(component.innerHTML).toMatchSnapshot();
  });
});
