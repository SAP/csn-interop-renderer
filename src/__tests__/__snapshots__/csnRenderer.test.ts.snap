// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom web-component Tests should properly render the provided CSN source document 1`] = `
"<p><strong>Primer:</strong></p>
<p><a href="https://sap.github.io/csn-interop-specification/spec-v1/csn-interop-effective" target="_blank">Core schema notation interoperability (short: CSN Interop)</a> is a powerful and flexible format used to describe entity and service models in the wider SAP and BTP ecosystem.
CSN files are JSON-based and provide comprehensive metadata about entities and their structure, relationships, and other aspects of the model.</p>
<h1 id="airline-example">Airline (Example)</h1>
<p>This CSN example document shows how the airline example is expressed with a CDS <strong>Service</strong> exposing the entities through an API.</p>
<h2 id="entity-definitions">Entity Definitions</h2>
<h3 id="airlineserviceairline">AirlineService.Airline</h3>
<p>Entity exposed via:
<a href="#airlineservice">AirlineService</a></p>
<p>Human readable description of the entity, in <strong>markdown</strong>.</p>

<p>@EndUserText.label: <code>"Airline"</code><br>@ObjectModel.modelingPattern: <code>{"#":"ANALYTICAL_DIMENSION"}</code></p>
<p>Elements: </p>
<table>
<tbody><tr><th>Element</th><th>Type</th><th>Description</th></tr><tr><td><strong id="airlineserviceairline-airlineid">AirlineID</strong><br><br></td><td><a href="#airlineuuid">AirlineUuid</a></td><td><p>Human readable description of the element, in <strong>markdown</strong>.</p>
<br>key: <code>true</code><br>@EndUserText.label: <code>"Airline"</code><br>@ObjectModel.text.element: <code>["Name"]</code></td></tr>
<tr><td><strong id="airlineserviceairline-name">Name</strong><br><br></td><td>cds.String(40)</td><td>@EndUserText.label: <code>"Name"</code><br>@Semantics.text: <code>true</code></td></tr>
<tr><td><strong id="airlineserviceairline-currencycode_code">CurrencyCode_code</strong><br><br></td><td>cds.String(3)</td><td>@EndUserText.label: <code>"Currency Code"</code><br>@Semantics.currencyCode: <code>true</code></td></tr>
</tbody></table>

<h3 id="airlineserviceairport">AirlineService.Airport</h3>
<p>Entity exposed via:
<a href="#airlineservice">AirlineService</a></p>
<p>@EndUserText.label: <code>"Airport"</code><br>@ObjectModel.modelingPattern: <code>{"#":"ANALYTICAL_DIMENSION"}</code></p>
<p>Elements: </p>
<table>
<tbody><tr><th>Element</th><th>Type</th><th>Description</th></tr><tr><td><strong id="airlineserviceairport-airportid">AirportID</strong><br><br></td><td>cds.String(3)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Airport"</code><br>@ObjectModel.text.element: <code>["Name"]</code></td></tr>
<tr><td><strong id="airlineserviceairport-name">Name</strong><br><br></td><td>cds.String(40)</td><td>@EndUserText.label: <code>"Name"</code><br>@Semantics.text: <code>true</code></td></tr>
<tr><td><strong id="airlineserviceairport-city">City</strong><br><br></td><td>cds.String(40)</td><td>@EndUserText.label: <code>"City"</code></td></tr>
<tr><td><strong id="airlineserviceairport-countrycode_code">CountryCode_code</strong><br><br></td><td>cds.String(3)</td><td>@EndUserText.label: <code>"Country Code"</code><br>@ObjectModel.foreignKey.association: <code>{"=":"to_CountryCode"}</code></td></tr>
<tr><td><strong id="airlineserviceairport-to_countrycode">to_CountryCode</strong><br><br></td><td>cds.Association</td><td>Association to one <a href="#airlineservicecountries">AirlineService.Countries</a> (path: <a href="#airlineservicecountries">AirlineService.Countries</a>.<a href="#airlineservicecountries-code">code</a>) via <a href="#airlineserviceairport-countrycode_code">CountryCode_code</a></td></tr>
</tbody></table>

<h3 id="airlineservicecountries">AirlineService.Countries</h3>
<p>Entity exposed via:
<a href="#airlineservice">AirlineService</a></p>
<p>@EndUserText.label: <code>"Countries"</code><br>@ObjectModel.modelingPattern: <code>{"#":"ANALYTICAL_DIMENSION"}</code></p>
<p>Elements: </p>
<table>
<tbody><tr><th>Element</th><th>Type</th><th>Description</th></tr><tr><td><strong id="airlineservicecountries-code">code</strong><br><br></td><td>cds.String(3)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Country Code"</code><br>@ObjectModel.text.association: <code>{"=":"texts"}</code></td></tr>
<tr><td><strong id="airlineservicecountries-texts">texts</strong><br><br></td><td>cds.Composition</td><td></td></tr>
</tbody></table>

<h3 id="airlineservicecountries_texts">AirlineService.Countries_texts</h3>
<p>Entity exposed via:
<a href="#airlineservice">AirlineService</a></p>
<p>@EndUserText.label: <code>"Countries Texts"</code><br>@ObjectModel.modelingPattern: <code>{"#":"LANGUAGE_DEPENDENT_TEXT"}</code></p>
<p>Elements: </p>
<table>
<tbody><tr><th>Element</th><th>Type</th><th>Description</th></tr><tr><td><strong id="airlineservicecountries_texts-code">code</strong><br><br></td><td>cds.String(3)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Country Code"</code></td></tr>
<tr><td><strong id="airlineservicecountries_texts-locale">locale</strong><br><br></td><td>cds.String(14)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Language Code"</code><br>@Semantics.language: <code>true</code></td></tr>
<tr><td><strong id="airlineservicecountries_texts-name">name</strong><br><br></td><td>cds.String(255)</td><td>@EndUserText.label: <code>"Name"</code><br>@Semantics.text: <code>true</code></td></tr>
<tr><td><strong id="airlineservicecountries_texts-descr">descr</strong><br><br></td><td>cds.String(1000)</td><td>@EndUserText.label: <code>"Description"</code><br>@Semantics.text: <code>true</code></td></tr>
</tbody></table>

<h3 id="airlineserviceflightconnection">AirlineService.FlightConnection</h3>
<p>Entity exposed via:
<a href="#airlineservice">AirlineService</a></p>
<p>@EndUserText.label: <code>"Flight Connection"</code><br>@ObjectModel.representativeKey: <code>{"=":"ConnectionID"}</code><br>@ObjectModel.modelingPattern: <code>{"#":"ANALYTICAL_DIMENSION"}</code></p>
<p>Elements: </p>
<table>
<tbody><tr><th>Element</th><th>Type</th><th>Description</th></tr><tr><td><strong id="airlineserviceflightconnection-airlineid">AirlineID</strong><br><br></td><td>cds.String(3)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Airline"</code><br>@ObjectModel.foreignKey.association: <code>{"=":"to_Airline"}</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-connectionid">ConnectionID</strong><br><br></td><td>cds.String(4)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Flight Number"</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-departureairport_airportid">DepartureAirport_AirportID</strong><br><br></td><td>cds.String(3)</td><td>@EndUserText.label: <code>"Departure Airport"</code><br>@ObjectModel.foreignKey.association: <code>{"=":"to_DepartureAirport"}</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-destinationairport_airportid">DestinationAirport_AirportID</strong><br><br></td><td>cds.String(3)</td><td>@EndUserText.label: <code>"Destination Airport"</code><br>@ObjectModel.foreignKey.association: <code>{"=":"to_DestinationAirport"}</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-departuretime">DepartureTime</strong><br><br></td><td>cds.Time</td><td>@EndUserText.label: <code>"Departure Time"</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-arrivaltime">ArrivalTime</strong><br><br></td><td>cds.Time</td><td>@EndUserText.label: <code>"Arrival Time"</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-distance">Distance</strong><br><br></td><td>cds.Integer</td><td>@EndUserText.label: <code>"Distance"</code><br>@Semantics.quantity.unitOfMeasure: <code>{"=":"DistanceUnit"}</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-distanceunit">DistanceUnit</strong><br><br></td><td>cds.String(3)</td><td>@EndUserText.label: <code>"Distance Unit"</code></td></tr>
<tr><td><strong id="airlineserviceflightconnection-to_airline">to_Airline</strong><br><br></td><td>cds.Association</td><td>Association to many <a href="#airlineserviceairline">AirlineService.Airline</a> (path: <a href="#airlineserviceairline">AirlineService.Airline</a>.<a href="#airlineserviceairline-airlineid">AirlineID</a>) via <a href="#airlineserviceflightconnection-airlineid">AirlineID</a></td></tr>
<tr><td><strong id="airlineserviceflightconnection-to_departureairport">to_DepartureAirport</strong><br><br></td><td>cds.Association</td><td>Association to many <a href="#airlineserviceairport">AirlineService.Airport</a> (path: <a href="#airlineserviceairport">AirlineService.Airport</a>.<a href="#airlineserviceairport-airportid">AirportID</a>) via <a href="#airlineserviceflightconnection-departureairport_airportid">DepartureAirport_AirportID</a></td></tr>
<tr><td><strong id="airlineserviceflightconnection-to_destinationairport">to_DestinationAirport</strong><br><br></td><td>cds.Association</td><td>Association to many <a href="#airlineserviceairport">AirlineService.Airport</a> (path: <a href="#airlineserviceairport">AirlineService.Airport</a>.<a href="#airlineserviceairport-airportid">AirportID</a>) via <a href="#airlineserviceflightconnection-destinationairport_airportid">DestinationAirport_AirportID</a></td></tr>
</tbody></table>

<h3 id="airlineserviceflight">AirlineService.Flight</h3>
<p>Entity exposed via:
<a href="#airlineservice">AirlineService</a></p>
<p>@EndUserText.label: <code>"Flight"</code><br>@ObjectModel.modelingPattern: <code>{"#":"ANALYTICAL_FACT"}</code><br>@ObjectModel.supportedCapabilities: <code>[{"#":"DATA_STRUCTURE"}]</code></p>
<p>Elements: </p>
<table>
<tbody><tr><th>Element</th><th>Type</th><th>Description</th></tr><tr><td><strong id="airlineserviceflight-airlineid">AirlineID</strong><br><br></td><td>cds.String(3)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Airline"</code><br>@ObjectModel.foreignKey.association: <code>{"=":"to_Airline"}</code></td></tr>
<tr><td><strong id="airlineserviceflight-flightdate">FlightDate</strong><br><br></td><td>cds.Date</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Flight Date"</code></td></tr>
<tr><td><strong id="airlineserviceflight-connectionid">ConnectionID</strong><br><br></td><td>cds.String(4)</td><td>key: <code>true</code><br>@EndUserText.label: <code>"Flight Number"</code><br>@ObjectModel.foreignKey.association: <code>{"=":"to_Connection"}</code></td></tr>
<tr><td><strong id="airlineserviceflight-price">Price</strong><br><br></td><td>cds.Decimal</td><td>precision: <code>16</code><br>scale: <code>3</code><br>@EndUserText.label: <code>"Price"</code><br>@Aggregation.default: <code>{"#":"MIN"}</code><br>@Semantics.amount.currencyCode: <code>{"=":"CurrencyCode_code"}</code></td></tr>
<tr><td><strong id="airlineserviceflight-currencycode_code">CurrencyCode_code</strong><br><br></td><td>cds.String(3)</td><td>@EndUserText.label: <code>"Currency Code"</code><br>@Semantics.currencyCode: <code>true</code></td></tr>
<tr><td><strong id="airlineserviceflight-planetype">PlaneType</strong><br><br></td><td>cds.String(10)</td><td>@EndUserText.label: <code>"Plane Type"</code></td></tr>
<tr><td><strong id="airlineserviceflight-maximumseats">MaximumSeats</strong><br><br></td><td>cds.Integer</td><td>@EndUserText.label: <code>"Maximum Seats"</code><br>@Aggregation.default: <code>{"#":"SUM"}</code></td></tr>
<tr><td><strong id="airlineserviceflight-occupiedseats">OccupiedSeats</strong><br><br></td><td>cds.Integer</td><td>@EndUserText.label: <code>"Occupied Seats"</code><br>@Aggregation.default: <code>{"#":"SUM"}</code></td></tr>
<tr><td><strong id="airlineserviceflight-to_airline">to_Airline</strong><br><br></td><td>cds.Association</td><td>Association to many <a href="#airlineserviceairline">AirlineService.Airline</a> (path: <a href="#airlineserviceairline">AirlineService.Airline</a>.<a href="#airlineserviceairline-airlineid">AirlineID</a>) via <a href="#airlineserviceflight-airlineid">AirlineID</a></td></tr>
<tr><td><strong id="airlineserviceflight-to_connection">to_Connection</strong><br><br></td><td>cds.Association</td><td>Association to many <a href="#airlineserviceflightconnection">AirlineService.FlightConnection</a> (path: <a href="#airlineserviceflightconnection">AirlineService.FlightConnection</a>.<a href="#airlineserviceflightconnection-airlineid">AirlineID</a>) via <a href="#airlineserviceflight-airlineid">AirlineID</a></td></tr>
</tbody></table>

<h3 id="unassignedentity">UnassignedEntity</h3>
<p>This entity is not prefixed with the service name and therefore not accessible through the service.</p>



<p>Elements: </p>
<table>
<tbody><tr><th>Element</th><th>Type</th><th>Description</th></tr><tr><td><strong id="unassignedentity-someelement">SomeElement</strong><br><br></td><td>cds.String</td><td>key: <code>true</code><br></td></tr>
</tbody></table>

<h2 id="type-definitions">Type Definitions</h2>
<h3 id="airlineuuid">AirlineUuid</h3>
<p>Description of the reuse type</p>

<table>
<tbody><tr><th>Type</th><th>Description</th></tr><tr><td>cds.String(3)</td><td>@EndUserText.label: <code>"Airline"</code><br>@ObjectModel.text.element: <code>["Name"]</code></td></tr>
</tbody></table>

<h2 id="services">Services</h2>
<h3 id="airlineservice">AirlineService</h3>
<p>This is describing the service that exposes the CDS entities through an API.</p>

<p>Exposed Entities:</p>
<ul>
<li><a href="#airlineserviceairline">AirlineService.Airline</a></li>
<li><a href="#airlineserviceairport">AirlineService.Airport</a></li>
<li><a href="#airlineservicecountries">AirlineService.Countries</a></li>
<li><a href="#airlineservicecountries_texts">AirlineService.Countries_texts</a></li>
<li><a href="#airlineserviceflightconnection">AirlineService.FlightConnection</a></li>
<li><a href="#airlineserviceflight">AirlineService.Flight</a></li>
</ul>
"
`;
